server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestCreateIndexAction.java

```diff
@@ -64,18 +64,36 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         }
 
         CreateIndexRequest createIndexRequest = new CreateIndexRequest(request.param("index"));
+
         if (request.hasContent()) {
-            Map<String, Object> sourceAsMap = XContentHelper.convertToMap(request.content(), false, request.getXContentType()).v2();
-            if (includeTypeName == false && sourceAsMap.containsKey("mappings")) {
-                Map<String, Object> newSourceAsMap = new HashMap<>(sourceAsMap);
-                newSourceAsMap.put("mappings", Collections.singletonMap(MapperService.SINGLE_MAPPING_NAME, sourceAsMap.get("mappings")));
-                sourceAsMap = newSourceAsMap;
-            }
+            Map<String, Object> sourceAsMap = XContentHelper.convertToMap(request.requiredContent(), false,
+                request.getXContentType()).v2();
+            sourceAsMap = prepareMappings(sourceAsMap, includeTypeName);
             createIndexRequest.source(sourceAsMap, LoggingDeprecationHandler.INSTANCE);
         }
+
         createIndexRequest.timeout(request.paramAsTime("timeout", createIndexRequest.timeout()));
         createIndexRequest.masterNodeTimeout(request.paramAsTime("master_timeout", createIndexRequest.masterNodeTimeout()));
         createIndexRequest.waitForActiveShards(ActiveShardCount.parseString(request.param("wait_for_active_shards")));
         return channel -> client.admin().indices().create(createIndexRequest, new RestToXContentListener<>(channel));
     }
+
+
+    static Map<String, Object> prepareMappings(Map<String, Object> source, boolean includeTypeName) {
+        if (includeTypeName || !source.containsKey("mappings")) {
```

Comment:
use == false for consistency with the rest of the codebase?

Reply:
Will do, this is proving a hard habit to break!